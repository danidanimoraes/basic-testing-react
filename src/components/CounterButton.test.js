import React from 'react';
import { shallow, mount, render } from 'enzyme';
import CounterButton from './CounterButton';
import "../../setupTests";


it('should shallow match snapshot Counterbutton', () => {
    const mockColor = 'red';
    expect(shallow(<CounterButton color={mockColor} />)).toMatchSnapshot()
});

it('should increment the counter', () => {
    const mockColor = 'red';
    const wrapper = shallow(<CounterButton color={mockColor} />);
    expect(wrapper.state()).toEqual({ count: 1 });
    wrapper.find('[id="counter"]').simulate('click');
    expect(wrapper.state()).toEqual({ count: 2 });
    expect(wrapper.props().color).toEqual(mockColor);
});