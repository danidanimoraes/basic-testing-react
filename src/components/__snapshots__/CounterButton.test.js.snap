// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should shallow match snapshot Counterbutton 1`] = `ShallowWrapper {}`;
